lib/bundler/dsl.rb

```diff
@@ -34,7 +34,9 @@ def initialize
     end
 
     def eval_gemfile(gemfile, contents = nil)
-      @gemfile = Pathname.new(gemfile)
+      expanded_gemfile_path = Pathname.new(gemfile).expand_path
+      @gemfile = expanded_gemfile_path
+      original_gemfile = @gemfile
```

Comment:
Yes, you're right. I think what this is telling me is that we need a spec specifically to cover this case. Though, I can't really think of a functionally useful test case. As in, a test with multiple Gemfiles that emulates a real-world use case. If you know any off the top of your head, let me know. Otherwise, I'll look around Github/think one up.